pkg/operator/render.go

```diff
@@ -68,30 +69,16 @@ func createDiscoveredControllerConfigSpec(infra *configv1.Infrastructure, networ
 	}
 
 	platform := "none"
-	//nolint:staticcheck
-	switch infra.Status.Platform {
-	case configv1.AWSPlatformType:
-		platform = "aws"
-	case configv1.AzurePlatformType:
-		platform = "azure"
-	case configv1.BareMetalPlatformType:
-		platform = "baremetal"
-	case configv1.GCPPlatformType:
-		platform = "gcp"
-	case configv1.OpenStackPlatformType:
-		platform = "openstack"
-	case configv1.LibvirtPlatformType:
-		platform = "libvirt"
-	case configv1.VSpherePlatformType:
-		platform = "vsphere"
+	infraPlatformString := string(infra.Status.PlatformStatus.Type)
+	if infraPlatformString != "" {
+		platform = strings.ToLower(infraPlatformString)
 	}
 
 	ccSpec := &mcfgv1.ControllerConfigSpec{
 		ClusterDNSIP:        dnsIP,
 		CloudProviderConfig: "",
 		EtcdDiscoveryDomain: infra.Status.EtcdDiscoveryDomain,
 		Platform:            platform,
-		Proxy:               &proxy.Status,
```

Comment:
You might want to rebase to get rid of this but there won't be conflicts anyway I guess

Reply:
done